artiq/language/environment.py

```diff
@@ -405,6 +420,30 @@ def get_dataset(self, key, default=NoDefault, archive=True):
             else:
                 return default
 
+    def get_dataset_metadata(self, key, default=NoDefault):
+        """Returns the metadata of a dataset.
+         
+        Returns dictionary with items describing the dataset, including the units, 
+        scale and precision. 
+
+        This function is used to get additional information for displaying the dataset.
+
+        Returns a dictionary which may contain zero or more of the items:
+        
+        'unit' - Physical unit associated with dataset.
```

Comment:
These should probably match the respective documentation in `set_dataset` (or just link to it).

Reply:
Linking to it sounds good.